src/common/fetch.rs

```diff
@@ -6,18 +6,31 @@ use async_std::path::PathBuf;
 use async_std::{fs, task};
 use bytes::Bytes;
 use reqwest::Client;
+use tokio::sync::Mutex;
 use tracing::{debug, trace};
 
 use crate::common::cache::cache_file_path;
 use crate::error;
 
+static FETCH_LOCK: Mutex<()> = Mutex::const_new(());
+
 pub async fn fetch_url(
     url: &str,
     attempts: u8,
     retry_sleep_ms: u64,
     retry_sleep_exponent_ms: u64,
 ) -> error::Result<Option<Bytes>> {
     trace!("fetch_url: url: {}", url);
+
+    let _guard = tokio::time::timeout(Duration::from_secs(10), FETCH_LOCK.lock())
```

Comment:
QQ since the mutex is wrapped in the timeout, is the _guard var actually needed or can it be 

```suggestion
    tokio::time::timeout(Duration::from_secs(10), FETCH_LOCK.lock())
```

since we're immediately awaiting on the returned future anyway

Reply:
We need the gaurd so that when this function is done, the gaurd gets droped and the lock is released. Because timeout tries to resolve the value of the function in this case the lock or fail. and we need to keep the lock till the function is done

Reply:
~~are you sure that's how it works? The whole promise returned from the timeout resolves in L26 due to the await. I might be wrong but I think if you want to hold a guard, to drop at the end of the fn, you need to pull out the FETCH_LOCK.lock() fn out of the timeout fn beacuse right now, the content of `_guard` is just a Promise~~
After reading the docs of timeout, i'm cool w/ this :)

Also, why are you calling it `gaurd`? once could be a typo but now i'm wondering if there's something i don't know